addons/account/models/account_payment.py

```diff
@@ -418,7 +418,7 @@ def _compute_payment_method_line_id(self):
             else:
                 pay.payment_method_line_id = False
 
-    @api.depends('payment_type', 'journal_id')
+    @api.depends('payment_type', 'journal_id', 'currency_id')
```

Comment:
You probably want to add currency_id in depends for AccountPaymentRegister (as you have the same compute)